sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs

```diff
@@ -0,0 +1,165 @@
+
+            var logger = loggerFactory.CreateLogger<LogsHelperTests>();
+
+            string log = "Hello from {name} {price}.";
+            logger.LogInformation(log, "tomato", 2.99);
+
+            var properties = new ChangeTrackingDictionary<string, string>();
+            LogsHelper.GetMessageAndSetProperties(processor.processedItems[0], properties);
+
+            Assert.True(properties.TryGetValue("name", out string name));
+            Assert.Equal("tomato", name);
+            Assert.True(properties.TryGetValue("price", out string price));
+            Assert.Equal("2.99", price);
+        }
+
+        [Fact]
+        public void PropertiesContainFieldsFromStructuredLogsIfParseStateValuesIsSet()
+        {
+            var processor = new TestLogProcessor();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.ParseStateValues = true;
+                    options.AddProcessor(processor);
+                });
+                builder.AddFilter(typeof(LogsHelperTests).FullName, LogLevel.Trace);
+            });
+
+            var logger = loggerFactory.CreateLogger<LogsHelperTests>();
+
+            logger.LogInformation("{Name} {Price}!", "Tomato", 2.99);
+
+            var properties = new ChangeTrackingDictionary<string, string>();
+            LogsHelper.GetMessageAndSetProperties(processor.processedItems[0], properties);
+
+            Assert.True(properties.TryGetValue("Name", out string name));
+            Assert.Equal("Tomato", name);
+            Assert.True(properties.TryGetValue("Price", out string price));
+            Assert.Equal("2.99", price);
+        }
+
+        [Fact]
+        public void PropertiesContainEventIdAndEventNameIfSetOnLog()
+        {
+            var processor = new TestLogProcessor();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.ParseStateValues = true;
+                    options.AddProcessor(processor);
+                });
+                builder.AddFilter(typeof(LogsHelperTests).FullName, LogLevel.Trace);
+            });
+
+            var logger = loggerFactory.CreateLogger<LogsHelperTests>();
+
+            EventId id = new EventId(1, "TestEvent");
+            logger.LogInformation(id, "Log Information");
+
+            var properties = new ChangeTrackingDictionary<string, string>();
+            LogsHelper.GetMessageAndSetProperties(processor.processedItems[0], properties);
+
+            Assert.True(properties.TryGetValue("EventId", out string eventId));
+            Assert.Equal("1", eventId);
+            Assert.True(properties.TryGetValue("EventName", out string eventName));
+            Assert.Equal("TestEvent", eventName);
+        }
+
+        internal class TestLogProcessor : BaseProcessor<LogRecord>
```

Comment:
Do we need this? OR can we re-use InMemoryExporter, and obtain the LogRecord from it?

Reply:
We don't. Was thinking to avoid the reference but we can use as we are using it for other packages as well.